CONTRIBUTING.md

```diff
@@ -0,0 +1,68 @@
+configurable is exposed in its Resource. Each configurable must be documented, so it is clear for a human operator when
+it should be used, and why the correct value cannot be guessed automatically.
+
+### Add new Feature Gate
+
+Think twice before you do. Feature gates make HCO very hard to test; each of them essentially duplicates our test
+matrix. They also complicate life for the human operator, who has to read the documentation to understand the
+implications of pressing a knob. You should add new featureGate only if hard-coding it to true considerably harms our
+typical users. Think if you can just add the new feature gate in a hard code manner to the requested operator.
+
+If there is a real need for a new feature gate, please follow these steps:
+
+1. In the PR message, describe the new feature gate, what it does and why it needed to be added to the HCO API.
+1. Add the new feature gate to the HyperConvergedFeatureGates struct
+   in [pkg/apis/hco/v1beta1/hyperconverged_types.go](pkg/apis/hco/v1beta1/hyperconverged_types.go)
+    - make sure the name of the feature gate field is as the feature gate field in the target operand, including casing.
+      It also must start with a capital letter, to be exposed from the api package.
+    - Set the field type to `FeatureGate`.
+    - Make sure the json name in the json tag is valid (e.g. starts with a small cap).
+    - add open-api annotations:
+        - add detailed description in the comment
+        - default annotation
+        - optional annotation
+
+for example:
+
+  ```golang
+  // Allow migrating a virtual machine with SRIOV interfaces.
+// When enabled virt-launcher pods of virtual machines with SRIOV
+// interfaces run with CAP_SYS_RESOURCE capability.
+// This may degrade virt-launcher security.
+// +optional
+// +kubebuilder:default=false
+SRIOVLiveMigration FeatureGate `json:"sriovLiveMigration,omitempty"`
+  ```
+
+1. Add `IsEnabled` method for the new feature gate. It should be something like this:
+
+  ```golang
+  func (fgs *HyperConvergedFeatureGates) IsHotplugVolumesEnabled() bool {
+return (fgs != nil) && (fgs.HotplugVolumes != nil) && (*fgs.HotplugVolumes)
+}
+  ```
+
+1. Add a set of unit tests
+   in [pkg/apis/hco/v1beta1/hyperconverged_types_test.go](pkg/apis/hco/v1beta1/hyperconverged_types_test.go)
+   to check this new function.
+1. Add the new feature gate to the default of the FeatureGates field in the `HyperConvergedSpec` struct
+1. Add the new feature gate to the relevant operator handler. Currently, this is only supported for KubeVirt. For
+   KubeVirt, do the following:
+   In [pkg/controller/operands/kubevirt.go](pkg/controller/operands/kubevirt.go)
+    - Add a constant for the feature gate name in the constant block marked with
+      the `// KubeVirt feature gates that are exposed in HCO API`
+      comment.
+    - Add the new feature gate and the new IsEnabled function to the map in the `getFeatureGateChecks` function.
+    
```

Comment:
I think we should also to mention how to regenerate all the manifests and the autogenerated code before submitting the PR